### DISCUSSION TITLE
Decide on gh-deploy

### DISCUSSION THREAD

**Discussion by pawamoy** [mkdocs/mkdocs#3793] [Ideas]
Two options:

- completely remove it (after deprecation period)
- move it as a third-party/sub-command (useful for initial work on subcommands, also see get-deps)

**Comment by pawamoy** [comment]
Related issues:

- https://github.com/mkdocs/mkdocs/issues/431
- https://github.com/mkdocs/mkdocs/issues/891 (do we keep gh-deploy?)
- https://github.com/mkdocs/mkdocs/issues/930 (what for?)
- https://github.com/mkdocs/mkdocs/issues/2065 (would break many plugins)
- https://github.com/mkdocs/mkdocs/issues/2792
- https://github.com/mkdocs/mkdocs/issues/2796
- https://github.com/mkdocs/mkdocs/issues/2891
- https://github.com/mkdocs/mkdocs/issues/3047
- https://github.com/mkdocs/mkdocs/issues/3190
- https://github.com/mkdocs/mkdocs/issues/3373
- https://github.com/mkdocs/mkdocs/issues/2144

**Comment by lovelydinosaur** [comment]
At this point in time I'd suggest we leave it as-is.
I'm not seeing sufficient motivation to drop this, right?

**Reply by Andre601** [reply]
I can't see a reason for this command to persist given there are much more powerful options available, such as GitHub Actions.
If anything could this command be transformed into a more general "Build site and Push to branch X" action, to be used on other git platforms and not just GitHub, because gh-deploy is pretty much this: A command to build the site and push to a specific branch.
